imnodes.cpp

```diff
@@ -1838,17 +1793,23 @@ static void MiniMapUpdate()
         mini_map_background = GImNodes->Style.Colors[ImNodesCol_MiniMapBackground];
     }
 
-    const ImRect& editor_rect = GImNodes->CanvasRectScreenSpace;
-
-    const ImVec2 editor_center(
-        0.5f * (editor_rect.Min.x + editor_rect.Max.x),
-        0.5f * (editor_rect.Min.y + editor_rect.Max.y));
+    // Create a child window bellow mini-map, so it blocks all mouse interaction on canvas.
+    int flags = ImGuiWindowFlags_NoBackground;
+    ImGui::SetCursorScreenPos(editor.MiniMapRectScreenSpace.Min);
+    ImGui::BeginChild("minimap", editor.MiniMapRectScreenSpace.GetSize(), false, flags);
+    ImGui::InvisibleButton("minimap", editor.MiniMapRectScreenSpace.GetSize());
 
-    const ImRect& mini_map_rect = GImNodes->MiniMapRectScreenSpace;
+    bool center_on_click =
+        editor.ClickInteraction.Type == ImNodesClickInteractionType_None &&
+        ImGui::IsItemActive() &&
+        ImGui::IsMouseDown(ImGuiMouseButton_Left);
+    if (center_on_click)
+    {
+        editor.ClickInteraction.Type = ImNodesClickInteractionType_CenterOnRequest;
+        editor.CenterOnRequest = MiniMapSpaceToGridSpace(editor, ImGui::GetMousePos());
```

Comment:
`editor.GridContentBounds` contains FLT_MAX and FLT_MIN here if there are no nodes. Thus `editor.CenterOnRequest` will contain FLT_MAX, and it propagates down to `editor.Panning` and to the node origin, causing a crash in ImGui when we try to render.

Reply:
We could probably skip centering completely when there is no content, as it has no perceivable effect.

Reply:
Disabled centering, when there's no content.

Also changed for GridContentBounds to be equal to the editor's canvas view if theres no content to avoid similar problems in the future.